[IMP] web: stop showing tracebacks to visitors

qsm-odoo opened the pull request:
Before this commit, any client error would lead to a displayed Odoo
traceback dialog. While the goal is obviously to have a code without any
possible traceback, visitors really should not see those.

This is especially true in the website: if a website visitor (who has no
idea the visited website uses Odoo) gets a traceback because of a Chrome
extension, it makes no sense to display the traceback at all. It can be
justified for connected users (not portal, but base.group_user) as they
likely are employees of the company using Odoo and should be able to see
issues and report them to us (e.g. even in the Chrome extension case: so
we can make our code support the breaking extension or decide that it
cannot be used alongside Odoo).

The errors for visitors are still logged in the browser console, even
with more detailed logs in that case.

Note that we use `await user.hasGroup('base.group_user')` upon error to
be able to implement this feature. In the frontend, it actually does a
RPC call as that group information is not in the session_info. We could
add it in the session_info, but since the use case is showing an error,
the extra RPC *upon error* is fine for now.

Notice that traceback dialogs (and the related error handling that comes
with them, like preventing an error to be considered as an error and
logged in the console) are also still enabled in debug mode or during
testing tours.

task-4290643

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/185481.png)](https://mergebot.odoo.com/odoo/odoo/pull/185481)

qsm-odoo commented:
Discussed with @ged-odoo already, I added an extra log for non-internal users in the end (important common info that only appeared in the traceback dialog and not in the console). Instead of pinging teams about this, I added it in the pad for the meeting of tomorrow. After that, if no strong objection, it's good to go IMO :+1: 

Note: the error service/dialog unit tests are already still passing. Waiting to see if full runbot green. Not sure if I should add a test to test that the not-connected users do not get a visual traceback... not sure how I would do that in a way that makes much sense.

ged-odoo commented:
indeed, i don't think that a test would add much value here

qsm-odoo reviewed (commented):
@robodoo r+

robodoo commented:
@qsm-odoo 'ci/runbot' failed on this reviewed PR.

qsm-odoo reviewed (commented):
@robodoo r+
